smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/TypeScriptDependency.java

```diff
@@ -157,11 +171,44 @@ public Symbol createSymbol(String name) {
     }
 
     /**
-     * Holds package-wide constants, such as the current version of the AWS SDK for JS v3.
+     * Reads the versions of AWS-published libraries from smithy-aws-typescript-codegen, if it's available
+     * on the classpath.
      */
     private static final class SdkVersion {
-        static final String LIVE = "3.18.0";
+        private static final Logger LOGGER = Logger.getLogger(SdkVersion.class.getName());
+        private static final String PROPERTIES_PATH =
+                "/software/amazon/smithy/aws/typescript/codegen/sdkVersions.properties";
+        private static final Map<String, String> VERSIONS;
+
+        static {
+            Map<String, String> tmpVersions;
+            try {
+                URL versionsUrl = SdkVersion.class.getResource(PROPERTIES_PATH);
+                if (versionsUrl == null) {
+                    throw new IOException();
+                }
+                Properties p = new Properties();
+                try (Reader r =
+                        new BufferedReader(new InputStreamReader(versionsUrl.openStream(), StandardCharsets.UTF_8))) {
+                    p.load(r);
+                }
+                final Map<String, String> versions = new HashMap<>(p.size());
+                p.forEach((k, v) -> {
+                    if (versions.put(k.toString(), v.toString()) != null) {
+                        throw new IllegalArgumentException(String.format("Multiple versions defined for %s", k));
+                    }
+                });
+                tmpVersions = Collections.unmodifiableMap(versions);
+            } catch (IOException e) {
+                LOGGER.info("Could not read AWS dependency versions from smithy-aws-typescript-codegen, "
+                        + "will use 'latest' for AWS dependencies");
+                tmpVersions = Collections.emptyMap();
+            }
+            VERSIONS = tmpVersions;
+        }
 
-        private SdkVersion() {}
+        private static String getVersion(String packageName) {
+            return VERSIONS.getOrDefault(packageName, "latest");
```

Comment:
This is very unfortunate as it essentially means you always need to have a dependency on the aws generator even if you're not using any aws stuff. That's probably still better than the current situation though

Reply:
Yeah I can't think of another alternative.